src/components/Layout/Nav.jsx

```diff
@@ -1,8 +1,24 @@
 import { Icon } from "@iconify/react";
 import header from "./../../../src/assets/apna-5.PNG";
 import { Link } from "react-router-dom";
+import { useData } from "../../context/data-context";
+import { useState } from "react";
+import { useAuthContext } from "../../context";
 
 const Nav = () => {
+    const { login, logOutUser } = useAuthContext()
+    console.log(login)
+    const { firstName } = login;
+    const [searchTxt, setSearchTxt] = useState("");
+    const {
+        state: { itemsInCart, itemsInWishlist }, dispatch } = useData();
+    const searchHandler = (e) => {
+        if (e.keyCode === 13) {
+            dispatch({ type: "SEARCH_PRODUCT", payload: searchTxt });
```

Comment:
try doing this search functionality from a useParams

Reply:
I will try